pkg/api/handlers/libpod/images.go

```diff
@@ -498,12 +502,74 @@ func CommitContainer(w http.ResponseWriter, r *http.Request) {
 	if len(query.Repo) > 0 {
 		destImage = fmt.Sprintf("%s:%s", query.Repo, tag)
 	}
-	commitImage, err := ctr.Commit(r.Context(), destImage, options)
+
+	if !query.Stream {
+		commitImage, err := ctr.Commit(r.Context(), destImage, options)
+		if err != nil && !strings.Contains(err.Error(), "is not running") {
+			utils.Error(w, http.StatusInternalServerError, fmt.Errorf("CommitFailure: %w", err))
+			return
+		}
+		utils.WriteResponse(w, http.StatusOK, entities.IDResponse{ID: commitImage.ID()})
+		return
+	}
+
+	// Channels all mux'ed in select{} below to follow API commit protocol
+	stdout := channel.NewWriter(make(chan []byte))
+	defer stdout.Close()
+	// Channels all mux'ed in select{} below to follow API commit protocol
+	options.CommitOptions.ReportWriter = stdout
+	var commitImage *libimage.Image
+	success := false
+	runCtx, cancel := context.WithCancel(r.Context())
+	go func() {
+		defer cancel()
+		commitImage, err = ctr.Commit(r.Context(), destImage, options)
+		if err == nil {
```

Comment:
I think we should preserve the error.  The error can be returned in case we didn´t stream yet.  The pull endpoints do that.

Reply:
Could write it to an error channel and use that in the switch table below.

Reply:
I do not know how to trigger a failure on this.  If a user specifies a bad container name, this is caught before the commit line. It would be a very strange occurance like OOM or OODisk space to cause this to fail.  Unless you have an idea.

Reply:
I'd try specifying some bogus change values:
```
podman (main) $ ./bin/podman commit --change BOGUS=FOO 123
Error: invalid change "BOGUS=FOO" - invalid instruction BOGUS
```

Reply:
Using this branch, I get the following output with podman-remote:
```
podman (commit) $ ./bin/podman-remote commit --change BOGUS=FOO 123
Error: 
```
With 4.7, I get:
```
podman (commit) $ /usr/bin/podman-remote commit --change BOGUS=FOO 123
Error: CommitFailure: invalid change "BOGUS=FOO" - invalid instruction BOGU
```

Reply:
Fixed